testing/set_library_versions.py

```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+"""
+This file is used by unit testing to specify library versions.
+Commands here should maintain alignment with the environment generated in 
+testing/Dockerfile.
+
+"""
+
+import json
+import sys
+import os
+
+# Read library versions from testcase library_versions.json file
+with open(os.path.join(sys.argv[1],'models','library_versions.json'),'r') as f:
+    library_versions = json.loads(f.read()) 
+
+# Checkout specified commit for each library
+for key in library_versions.keys():
+    if key == "IBPSA_COMMIT":
+        os.system('cd $MODELICAPATH/IBPSA && git checkout {}'.format(library_versions[key]))
```

Comment:
To avoid code injection you can add double quotes around the curly braces, see https://stackoverflow.com/questions/15783701/which-characters-need-to-be-escaped-when-using-bash

Reply:
Sorry I'm not sure what you mean from the link. Can you quckly type what you have in mind? no need to be perfect, just to see the idea.

Reply:
I think use:
```
os.system('cd $MODELICAPATH/IBPSA && git checkout "{}"'.format(library_versions[key]))
```
so that the contents of the bracket is always a string argument of ``git checkout`` rather than potentially unwanted additional code and commands.

Reply:
yup

Reply:
aha, I see now. Implemented here https://github.com/ibpsa/project1-boptest/pull/223/commits/c50e790076479a8ac0dddc6ddaabb1af5434b741.